File: examples/review_diff_decision_test.py
from review_diff_decision import decide_review
from review_diff_forma import ReviewDiffFinding, ReviewDiffOutput


def test_approves_clean_reviewed_output() -> None:
    output = ReviewDiffOutput(summary="No issues.", findings=[], clean=True)

    assert decide_review(output) == {
        "status": "approve",
        "finding_count": 0,
        "paths": [],
        "summary": "No issues.",
    }


def test_requests_changes_and_preserves_affected_paths() -> None:
    output = ReviewDiffOutput(
        summary="Bounds check is missing.",
        findings=[
            ReviewDiffFinding(path="src/review.py", line=42, message="Validate the array index before access."),
            ReviewDiffFinding(path="src/review.py", message="Add a regression test for empty input."),
            ReviewDiffFinding(path="test/test_review.py", message="Cover empty input."),
        ],
        clean=False,
    )

    assert decide_review(output) == {
        "status": "request_changes",
        "finding_count": 3,
        "paths": ["src/review.py", "test/test_review.py"],
        "summary": "Bounds check is missing.",
    }


if __name__ == "__main__":
    test_approves_clean_reviewed_output()
    test_requests_changes_and_preserves_affected_paths()

File: examples/review_diff_decision.test.ts
import { describe, expect, it } from "vitest";
import { decideReview } from "./review_diff_decision.js";
import type { ReviewDiffOutput } from "./review_diff.forma.js";

describe("review_diff host decision workflow", () => {
  it("approves clean reviewed output", () => {
    const output: ReviewDiffOutput = {
      summary: "No issues.",
      findings: [],
      clean: true,
    };

    expect(decideReview(output)).toEqual({
      status: "approve",
      findingCount: 0,
      paths: [],
      summary: "No issues.",
    });
  });

  it("requests changes and preserves affected paths", () => {
    const output: ReviewDiffOutput = {
      summary: "Bounds check is missing.",
      findings: [
        { path: "src/review.ts", line: 42, message: "Validate the array index before access." },
        { path: "src/review.ts", message: "Add a regression test for empty input." },
        { path: "test/review.test.ts", message: "Cover empty input." },
      ],
      clean: false,
    };

    expect(decideReview(output)).toEqual({
      status: "request_changes",
      findingCount: 3,
      paths: ["src/review.ts", "test/review.test.ts"],
      summary: "Bounds check is missing.",
    });
  });
});
